package com.metasoft.node;
option cc_generic_services = true;
option java_generic_services = true;
option java_package = "com.metasoft.flying.node";
option java_outer_classname = "FlightProto";
option py_generic_services = true;

message PvpSearchRequest {
  required int64 uid = 1;
  required int32 gold = 2;
}

message PvpSearchResponse {
  optional bool ok = 1 [default=false];
  optional string msg = 2;
  required int64 uid = 3;
}

message PvpBeginRequest {
  repeated int64 uidSeq = 1;
  required int32 gold = 2;
}

message PvpResponse {
  optional bool ok = 1 [default=false];
  optional string msg = 2;
}


service PvpSearchService {
  rpc search (PvpSearchRequest) returns (PvpSearchResponse);
  rpc begin (PvpBeginRequest) returns (PvpResponse);
}

